### ISSUE TITLE
Number of shots in Fock and Gaussian spaces

### ISSUE THREAD

**Issue by kareem1925** [XanaduAI/strawberryfields#333]
#### Issue description

```python
import strawberryfields as sf
from strawberryfields.ops import *
import numpy as np
sf.__version__  
>> '0.12.1'

#### create random points

x1 = 1
x2 = 0

eng = sf.Engine(
    backend="fock",
    backend_options={"cutoff_dim": 8,'shots':5000}
)
dist = sf.Program(4)

with dist.context as q:
    Fock(1) |q[0]
    Vacuum() | q[1]
    Dgate(a=x1) | q[2]
    Dgate(a=x2) | q[3]
    BSgate() | (q[0],q[1])
    BSgate() | (q[2],q[3])
    CZgate(np.pi) | (q[1],q[3])
    BSgate() | (q[0],q[1])
    BSgate() | (q[2],q[3])
    MeasureFock() | q[0]
    MeasureFock() | q[1]

results = eng.run(dist)
print(results.samples)

>> [0, 0, None, None]
```

The code above is an implementation of this figure:
![image](https://user-images.githubusercontent.com/30449261/77422926-f3447400-6dd6-11ea-9e85-2882df01cf39.png)

I just want to know if I’m correct with my code and also how I can interpret the outcome as stated in the paper. I also want to know why the number of the samples is just one and I set the shots to 5000

According to this [discussion](https://discuss.pennylane.ai/t/swap-test-in-using-cv/365/3) I am requesting an explanation for the number of shots in both the gaussian and fock spaces. 
This is the [paper](https://arxiv.org/pdf/1303.6814.pdf 
) I'm trying to implement

**Comment by co9olguy** [comment]
Thanks @kareem1925. This is a good spot for us to discuss any potential bugs in the SF code (i.e., if you're not seeing the right number of shots that you expect). We can't guarantee any support for research questions here though. For the benefit of future viewers, can you streamline your issue to focus up-front on the potential observed bug? Thanks :smile:

**Comment by josh146** [comment]
Hi @kareem1925, the number of shots to run a program with are runtime options, so they are passed in `eng.run()`, not when creating the backend:

```python
results = eng.run(dist, run_options={"shots": 5000})
```

Unfortunately, the Fock backend does not currently support multiple shots! This is something we are working to support, and hopefully should have this available soon.

**Comment by thisac** [comment]
Closed in favour of #678.
